### ISSUE TITLE
crash when minimizing the window

### ISSUE THREAD

**Issue by jice-nospam** [thebracket/bracket-lib#81]
I've upgraded bracketlib to version aa37 and still got the error when minimizing the window : 

thread 'main' panicked at 'attempt to divide by zero', C:\.cargo\registry\src\github.com-1ecc6299db9ec823\bracket-terminal-0.7.0\src\bterm.rs:155:17

My cargo.lock:
name = "bracket-lib"
version = "0.7.0"
source = "git+https://github.com/thebracket/bracket-lib#aa37c94de31f9902a0e1c92089f2150e6cc52da7"

**Comment by thebracket** [comment]
Oops; I put a patch into Cargo for the RLTK lib, not the bracket-lib. Try this in your `cargo.toml`:

```toml
# Remove before publish!
[patch.crates-io]
bracket-terminal = { git = "https://github.com/thebracket/bracket-lib/" }
```

**Comment by jice-nospam** [comment]
I can't get the fix while depending on the git version of bracketlib.
I think the git version of bracketlib should have local dependencies :
bracket-terminal={path="./bracket-terminal"}
instead of pointing to crates.io. Versions should be hardcoded only before publishing on crates.io.

**Comment by thebracket** [comment]
On my local build, this appears fixed - it doesn't crash when minimized. The `astar_mouse` example in `bracket-terminal` is the best repro I've found; it crashed before the fix, and doesn't crash now.

**Comment by jice-nospam** [comment]
It works!

**Comment by imoea** [comment]
While it doesn't for the `astar_mouse` example, the window still crashes when minimizing with the latest commit, like in the following:
```rust
use bracket_terminal::prelude::*;

struct State {}

impl GameState for State {
    fn tick(&mut self, ctx: &mut BTerm) {
        let _mouse_p = ctx.mouse_point();
    }
}

fn main() -> BError {
    let context = BTermBuilder::simple80x50().build()?;
    let gs = State {};
    main_loop(context, gs)
}
```
thread 'main' panicked at 'attempt to divide by zero', bracket-terminal\src\bterm.rs:176:17

**Comment by jice-nospam** [comment]
I reproduce it but I don't get the same error. For me the crash occurs in this function :
```rust
  #[inline]
    #[must_use] pub fn new<T>(x: T, y: T) -> Point
    where
        T: TryInto<i32>,
    {
        Point {
            x: x.try_into().ok().unwrap(),
            y: y.try_into().ok().unwrap(),
        }
    }
```
because the input values cannot be converted to i32. They come from Winit's WindowEvent::Moved event, which stores thoses values as f64.
This would fix the crash but not sure silently ignoring the error is the right way to fix it :
```rust
            x: x.try_into().ok().unwrap_or(0),
            y: y.try_into().ok().unwrap_or(0),
```

**Comment by jice-nospam** [comment]
Ok the point conversion error occurs on the input_api branch while the divide by zero error is on the master. The divide by zero correction is only on the input_api branch right now so this makes sense it still crashes on master.

The mouse coordinates conversion crash has problably come with the latest input commits.

**Comment by jice-nospam** [comment]
I fixed it in the commit 8508d92 in this [pull request](https://github.com/thebracket/bracket-lib/pull/95). 
Apparently, when the window is minimized, mouse position returns non valid coordinates (maybe NaN ?) which made the conversion to i32 crash.
